src/release/breaking-changes/win_lifecycle_process_function.md

```diff
@@ -0,0 +1,42 @@
+---
+title: Introduction of FlutterEngine::ProcessExternalWindowMessage
+description: External windows should call ProcessExternalWindowMessage to be considered for application lifecycle events
+---
+
+## Summary
+
+In order to be included in the logic for app lifecycle on Windows, any
+external windows added to a Flutter app should invoke
+`FlutterEngine::ProcessExternalWindowMessage` from its `WndProc` function.
+
+## Description of change
+
+Implementing application lifecycle on Windows involves listening for Window
+messages in order to update the lifecycle state. In order for additional
+non-Flutter windows to affect the lifecycle state, they must forward their
+window messages to `FlutterEngine::ProcessExternalWindowMessage` from their
+`WndProc` functions. This function returns an `std::optional<LRESULT>`, which
+is `std::nullopt` when the message is received, but not consumed. When the
+returned result has a value, the message has been consumed, and further
+processing in `WndProc` should cease.
+
+## Migration guide
+
+An example `WndProc` procedure that invokes
+`FlutterEngine::ProcessExternalWindowMessage` is provided below:
+
+```
+CALLBACK LRESULT WndProc(HWND hwnd, UINT msg, WPARAM wparam, LPARAM lparam) {
+    std::optional<LRESULT> result = FlutterEngine::ProcessExternalWindowMessage(hwnd, msg, wparam, lparam);

```

Comment:
`FlutterEngine::ProcessExternalWindowMessage` isn't a static method, right? Most apps probably have a `FlutterViewController` member in their Flutter window ([example](https://github.com/flutter/flutter/blob/5649161bd7fc591e313c12f35fccfe4c8e211dbe/examples/hello_world/windows/runner/flutter_window.h#L34C28-L34C49)) and should so something like:

```
std::optional<LRESULT> result = flutter_controller_->engine()->ProcessExternalWindowMessage(hwnd, msg, wparam, lparma);
```